lib/data/managers/firestore_manager.dart

```diff
@@ -140,12 +140,32 @@ class FireStoreManager {
             .then((featuredArticlesSnapshot) async {
           if (featuredArticlesSnapshot.data != null &&
               featuredArticlesSnapshot.data[documentFieldStatus] == DataStatus.PUBLISHED) {
-            listOfFeaturedArticles.add(
-                ArticleEntity.articleFromDocument(featuredArticlesSnapshot));
+            listOfFeaturedArticles.add(ArticleEntity.articleFromDocument(featuredArticlesSnapshot));
           }
         });
       }
     }
     return listOfFeaturedArticles;
   }
+
+  Future<ArticleEntity> getRelatedArticles(ArticleEntity selectedArticle) async {
+      if (selectedArticle.relatedArticlesPathReference != null) {
+          selectedArticle.relatedArticles.clear();
+          for (var relatedArticlesPathReference in selectedArticle
+              .relatedArticlesPathReference) {
+            await Firestore.instance
```

Comment:
looks like we repeat this code to get a published document in a few methods, could it be extracted?
Out of interest is it worth trying to run the status filter on the firestore query rather than getting the data then checking status?
```
firestore.collection(collection)
.where(id == docId)
.where(status == published)
```

Reply:
Now the checking of Published status is extracted into a method. I'll try to refactor with a new filter method in the future.